system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js

```diff
@@ -25,3 +22,34 @@ test_newtab({
     }
   }
 });
+
+test_newtab({
+  before: setDefaultTopSites,
+  // Test verifies that the next top site in queue replaces a dismissed top site.
+  test: async function defaultTopSites_dismiss() {
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-icon"),
+      "Topsite tippytop icon not found");
+
+    let defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
+    Assert.equal(defaultTopSitesNumber, 6, "6 top sites are loaded by default");
+
+    let secondTopSite = content.document.querySelector(".top-sites-list li:nth-child(2) a").getAttribute("href");
+
+    let contextMenuItems = content.openContextMenuAndGetOptions(".top-sites-list li:first-child");
+    Assert.equal(contextMenuItems[4].textContent, "Dismiss", "'Dismiss' is the 5th item in the context menu list");
+
+    contextMenuItems[4].querySelector("a").click();
+
+    // Need to wait for dismiss action.
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-sites-list li:first-child a").getAttribute("href") === secondTopSite,
+      "First topsite was dismissed");
+
+    defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
+    Assert.equal(defaultTopSitesNumber, 5, "5 top sites are displayed after one of them is dismissed");
+  },
+  async after() {
+    await SpecialPowers.pushPrefEnv({set: [["browser.newtabpage.blocked", ""]]});
```

Comment:
This doesn't seem right to call `SpecialPowers` here. There's already a `pushPrefs` globally, but the `before` should actually be using the scoped one instead. The scoped one cleans up after a test runs whereas the global one cleans up only when the test file is completed I believe.

Reply:
Sure, but using the scoped one from `before` should clear that pref when the test finishes.